workflow/artifacts/s3/s3.go

```diff
@@ -155,6 +156,23 @@ func (s3Driver *ArtifactDriver) Save(path string, outputArtifact *wfv1.Artifact)
 	return err
 }
 
+// Delete deletes an artifact from an S3 compliant storage
+func (s3Driver *ArtifactDriver) Delete(artifact *wfv1.Artifact) error {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
minor - but @juliev0 and I found out this context is wrong, it should not have a cancel, it should just be `context.TODO()` - minor - so only